### PR TITLE
[MENFORCER-372] Drop PlexusTestCase and junit:junit dependency

### PR DESCRIPTION
**Pull request by timtebeek**
Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MENFORCER) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MENFORCER-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MENFORCER-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

----

Turns out it was way easier to remove PlexusTestCase than I had originally thought. :sweat_smile:

### CODE REVIEW

**Comment by timtebeek** [comment]
Hmm the additional test failures are somewhat puzzling; needs a closer look, which in my case is limited to evenings/weekends. :disappointed:

**Comment by timtebeek** [comment]
Turns out we did need the junit-jupiter-engine [after all](https://github.com/apache/maven-enforcer/pull/113#discussion_r749185199); Time for another quick review @slawekjaranowski ?

File: pom.xml [hunk 1]
```diff
@@ -128,22 +128,15 @@
...
-        <artifactId>junit</artifactId>
-        <version>4.13.2</version>
-        <scope>test</scope>
-      </dependency>
-      <dependency>
-        <groupId>org.mockito</groupId>
-        <artifactId>mockito-core</artifactId>
-        <version>4.0.0</version>
+        <groupId>org.junit.jupiter</groupId>
+        <artifactId>junit-jupiter-engine</artifactId>
```
Comment (slawekjaranowski) [inline pom.xml:132]: Is it needed?

**Review by slawekjaranowski — COMMENTED** [review]
Did you run?
```
mvn dependency:tree 
mvn dependency:analyze
```

And are you sure that `junit4` is not available as transitive dependency?

File: pom.xml (hunk 1 above)
Comment (timtebeek) [reply pom.xml:132]: Without `junit-jupiter-engine` the tests fail in both `enforcer-rules` and `maven-enforcer-plugin`.

I had run `dependency:tree`, which shows JUnit 4 is no longer on the classpath.

`dependency:analyze` shows `enforcer-rules` would only need `mockito-core` as opposed to `mockito-junit-jupiter`, but then you'd have to manage that dependency as well.

**Comment by slawekjaranowski** [comment]
hint add to `pluginManagment` in root pom
```xml
        <plugin>
          <groupId>org.apache.maven.plugins</groupId>
          <artifactId>maven-surefire-plugin</artifactId>
          <version>3.0.0-M5</version>
        </plugin>
```

**Comment by timtebeek** [comment]
Done! That indeed helps, and seems a nice addition to not have to declare the engine anymore.

Would you want to restore the `mockito-core` dependency + management? Keep in mind that would trigger Dependabot twice, or require a version variable.

**Comment by slawekjaranowski** [comment]
For many components from the same project please use property as version and will be one update by dependabot.

Please also check if `maven-plugin-testing-harness`, `maven-compact` is still needed.

**Comment by timtebeek** [comment]
Went through the `mvn dependency:analyze` output once more and removed all unused declared dependencies.

**Comment by jira-importer** [comment]
Resolve #488

**Pull request closed without merging**